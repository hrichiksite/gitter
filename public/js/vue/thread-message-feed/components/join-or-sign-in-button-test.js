const mount = require('../../__test__/vuex-mount');
const { default: JoinOrSignInButton } = require('./join-or-sign-in-button.vue');
const {
  createSerializedRoomFixture,
  createSerializedUserFixture
} = require('../../__test__/fixture-helpers');

describe('thread-message-feed join-or-sign-in-button', () => {
  const addRoomAndUserToStore = (githubOnlyRoom = false) => store => {
    store.state.isLoggedIn = true;
    store.state.user = createSerializedUserFixture({ providers: ['gitlab'] });
    const providers = githubOnlyRoom ? { providers: ['github'] } : {};
    const room = createSerializedRoomFixture('test', { ...providers, roomMember: false });
    store.state.displayedRoomId = room.id;
    store.state.roomMap = { [room.id]: room };
  };

  describe('user allowed to join room', () => {
    it('matches snapshot', () => {
      const { wrapper } = mount(JoinOrSignInButton, {}, addRoomAndUserToStore());
      expect(wrapper.element).toMatchSnapshot();
    });

    it('matches snapshot when loading', () => {
      const { wrapper } = mount(JoinOrSignInButton, {}, store => {
        addRoomAndUserToStore()(store);
        store.state.joinRoomRequest = { loading: true, error: false };
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('matches snapshot when request failed', () => {
      const { wrapper } = mount(JoinOrSignInButton, {}, store => {
        addRoomAndUserToStore()(store);
        store.state.joinRoomRequest = {
          loading: false,
          error: new Error('Internal Server Error')
        };
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('joinRoom action called on click', () => {
      const { wrapper, stubbedActions } = mount(JoinOrSignInButton, {}, addRoomAndUserToStore());
      wrapper.find('.chat-input__btn').trigger('click');

      expect(stubbedActions.joinRoom).toHaveBeenCalledWith(expect.anything(), undefined, undefined);
    });
  });

  it('user not allowed to join - matches snapshot', () => {
    const { wrapper } = mount(JoinOrSignInButton, {}, addRoomAndUserToStore(true));
    expect(wrapper.element).toMatchSnapshot();
  });

  it('shows sign in button for nli users', () => {
    const { wrapper } = mount(JoinOrSignInButton, {}, store => {
      store.state.isLoggedIn = false;
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
