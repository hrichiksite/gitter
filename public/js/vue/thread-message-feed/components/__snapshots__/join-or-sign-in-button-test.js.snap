// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`thread-message-feed join-or-sign-in-button shows sign in button for nli users 1`] = `
<footer
  class="chat-input chat-input-nli"
>
  <a
    class="chat-input__btn"
    href="/login?action=login"
    target="_top"
  >
    
    Sign in to start talking
  
  </a>
</footer>
`;

exports[`thread-message-feed join-or-sign-in-button user allowed to join room matches snapshot 1`] = `
<footer
  class="chat-input chat-input-nli"
>
  <div
    class="join-button-container"
  >
    <button
      class="chat-input__btn"
    >
      
      Join room
    
    </button>
     
    <!---->
     
    <!---->
  </div>
</footer>
`;

exports[`thread-message-feed join-or-sign-in-button user allowed to join room matches snapshot when loading 1`] = `
<footer
  class="chat-input chat-input-nli"
>
  <div
    class="join-button-container"
  >
    <button
      class="chat-input__btn"
    >
      
      Join room
    
    </button>
     
    <!---->
     
    <loading-spinner-stub
      class="join-room-loading-icon"
    />
  </div>
</footer>
`;

exports[`thread-message-feed join-or-sign-in-button user allowed to join room matches snapshot when request failed 1`] = `
<footer
  class="chat-input chat-input-nli"
>
  <div
    class="join-button-container"
  >
    <button
      class="chat-input__btn"
    >
      
      Join room
    
    </button>
     
    <span
      class="error-text error-box"
    >
      
      Joining room failed: Internal Server Error
    
    </span>
     
    <!---->
  </div>
</footer>
`;

exports[`thread-message-feed join-or-sign-in-button user not allowed to join - matches snapshot 1`] = `
<footer
  class="chat-input chat-input-nli"
>
  <div
    class="not-allowed-to-join"
  >
    
    Only GitHub users can join this room.
  
  </div>
</footer>
`;
